engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -2563,13 +2635,27 @@ private void checkVectorizability(@NotNull final MethodCallExpr n,
         // expression evaluation code that expects singular values. This check makes sure that numba/dh vectorized
         // functions must be used alone as the entire expression after removing the enclosing parentheses.
         Node n1 = n;
+        boolean autoCastChecked = false;
         while (n1.hasParentNode()) {
             n1 = n1.getParentNode().orElseThrow();
             Class<?> cls = n1.getClass();
 
             if (cls == CastExpr.class) {
-                throw new PythonCallVectorizationFailure(
-                        "The return values of Python vectorized function can't be cast: " + n1);
+                if (!autoCastChecked && n.getData(QueryLanguageParserDataKeys.PY_CALLABLE_DETAILS).isCasted()) {
+                    autoCastChecked = true;
+                } else {
+                    throw new PythonCallVectorizationFailure(
+                            "The return values of Python vectorized function can't be cast: " + n1);
+                }
+            } else if (cls == MethodCallExpr.class) {
+                String methodName = ((MethodCallExpr) n1).getNameAsString();
+                if (!autoCastChecked && n.getData(QueryLanguageParserDataKeys.PY_CALLABLE_DETAILS).isCasted()
+                        && methodName.endsWith("Cast")) {
```

Comment:
Is there any chance that a python function with `Cast` at the end of the name will trigger weird behavior?  Does it need to be tested?

Reply:
I don't think so. The `isCasted()` condition guarantees that it is the artificially inserted the `...Cast` method that we are checking.